src/test_application/OSPMagnum.cpp

```diff
@@ -45,21 +46,51 @@ OSPMagnum::OSPMagnum(const Magnum::Platform::Application::Arguments& arguments,
 {
     //.setWindowFlags(Configuration::WindowFlag::Hidden)
 
+    // Initialize ImGui
+    m_imgui = Magnum::ImGuiIntegration::Context(
+        Magnum::Vector2{windowSize()} / dpiScaling(),
+        windowSize(), framebufferSize());
+    m_implot = ImPlot::CreateContext();
+
     m_timeline.start();
 
 }
 
 OSPMagnum::~OSPMagnum()
 {
+    // Free ImPlot
+    // ImPlot is not part of Magnum's imgui integration, and is handled manually
+    ImPlot::DestroyContext(m_implot);
```

Comment:
@jonesmz suggested a `unique_ptr` with a custom deleter

Reply:
Could you remind me what the context was?

The point of having a unique_ptr with a custom deleter is that when the unique_ptr goes out of scope the custom deleter gets triggered, which can either delete the thing, or cause whatever arbitrary outcomes to happen.

Reply:
ImGui's context is managed by OSPMagnum since it's a magnum plugin that integrates seamlessly. ImPlot is an external library, so its context needs to be manually freed. You suggested the `unique_ptr` so that we can use the default `~OSPMagnum`.